components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -352,18 +395,30 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  CGFloat availableWidth = CGRectGetWidth(self.bounds);
-  CGFloat requiredWidth = [self justifiedWidth];
-  BOOL canBeJustified = availableWidth >= requiredWidth;
-  self.containerView.distribution = canBeJustified ? UIStackViewDistributionFillEqually
-                                                   : UIStackViewDistributionFillProportionally;
+  [self updateStackviewLayout];
 
+  if (self.selectionIndicatorView) {
+    [self updateSelectionIndicatorToIndex:[self.items indexOfObject:self.selectedItem]];
+  }
   if (!self.initialScrollDone) {
     self.initialScrollDone = YES;
     [self scrollUntilSelectedItemIsVisibleWithoutAnimation];
   }
 }
 
+- (void)updateStackviewLayout {
+  CGFloat availableWidth = CGRectGetWidth(self.bounds);
+  CGFloat requiredWidth = [self justifiedWidth];
+  UIStackViewDistribution distribution = availableWidth >= requiredWidth
+                                             ? UIStackViewDistributionFillEqually
+                                             : UIStackViewDistributionFillProportionally;
+  if (self.containerView.distribution != distribution) {
+    self.containerView.distribution = distribution;
+    [self.containerView setNeedsLayout];
+  }
+  [self.containerView layoutIfNeeded];
```

Comment:
It's not the best idea, but seems not to cause any errors during testing. Unless we switch to fully manual layout, I'm not sure how else I can get the stack view to reposition its subviews.  I know it's possible to "estimate" the position of those views, but I prefer not to call that code if we can avoid it.

Let me see what breaks when I remove this and I will post back here.

Reply:
According to apple, this method will call LayoutSubviews which could create a calling cycle. The reason that the tests not broken probably because the 

> If no layout updates are pending, this method exits without modifying the layout or calling any layout-related callbacks.

Reply:
Calling it from the same view is definitely bad. Calling it from another view is not great.  However, it's not returning while doing nothing - it fixes the layout bugs when I test on the simulator.

I definitely need a hand figuring-out how to get the indicator to appear at the correct location. Forcing a subview to layout isn't the worse option, but it's far from ideal.